git_machete/git_operations.py

```diff
@@ -581,7 +581,7 @@ def get_reflog(self, branch: AnyBranchName) -> List[GitReflogEntry]:
             return self.__reflogs_cached[branch]
 
     def create_branch(self, branch: LocalBranchShortName, out_of_revision: AnyRevision) -> None:
-        self._run_git("checkout", "-b", branch, out_of_revision)
+        self._run_git("branch", branch, out_of_revision)
```

Comment:
Whoops this will affect quite a few places in an undesired way probably 😅

Reply:
@PawelLipski Yeah, i was just testing if this indeed was causing branches to be switched. I think additional flag `switch_branch` should do the trick here

Reply:
Better add a separate method if possible!

Reply:
@PawelLipski I think implementing it in separate method is not easily doable since this method is used by another method in at least 2 different contexts so i will still need to add some flag in order to switch between functions, wdyt?

Reply:
Hmmm okay let's leave a flag indeed 🤔